ipa-core/src/protocol/ipa_prf/malicious_security/hashing.rs

```diff
@@ -0,0 +1,171 @@
+use std::convert::Infallible;
+
+use generic_array::GenericArray;
+use sha2::{
+    digest::{Output, OutputSizeUser},
+    Digest, Sha256,
+};
+
+use crate::{
+    ff::{Field, Serializable},
+    protocol::prss::FromRandomU128,
+};
+
+#[derive(Debug, PartialEq)]
+pub struct Hash(Output<Sha256>);
+
+impl Serializable for Hash {
+    type Size = <Sha256 as OutputSizeUser>::OutputSize;
+
+    type DeserializationError = Infallible;
+
+    fn serialize(&self, buf: &mut GenericArray<u8, Self::Size>) {
+        buf.copy_from_slice(&self.0);
+    }
+
+    fn deserialize(buf: &GenericArray<u8, Self::Size>) -> Result<Self, Self::DeserializationError> {
+        Ok(Hash(*Output::<Sha256>::from_slice(buf)))
```

Comment:
for the sake of completeness, we should cover the deserialization phase with a unit test too. Maybe just generate a random hash and run it through serde

Reply:
OK, can do.